src/query/storage/m3/storage.go

```diff
@@ -197,6 +200,13 @@ func (s *m3storage) FetchCompressed(
 		return result, noop, err
 	}
 
+	if processor := queryOptions.IterationOptions.SeriesIteratorProcessor; processor != nil {
+		if err := processor.InspectSeries(ctx, result.SeriesIterators()); err != nil {
+			accumulator.Close()
+			return result, noop, err
```

Comment:
Inspecting the series should never error query execution, instead logging and ignoring is fine